R/textanal.R

```diff
@@ -0,0 +1,229 @@
+  tr="turkish",
+  uk="ukrainian",
+  vi="vietnamese")
+
+# Guess language of the text based on the first 10 rows of it.
+# Used to give default for get_stopwords().
+guess_lang_for_stopwords <- function(text) {
+  text <- head(text, 10)
+  lang_code <- get_mode(cld3::detect_language(text))
+  if (is.na(lang_code)) { # cld3 could not guess language.
+    return("english") # Default to English
+  }
+  lang_name <- lang_code_mapping[lang_code]
+  if (is.na(lang_name)) { # The language cld3 returned is not supported by get_stopwords().
+    return("english") # Default to English
+  }
+  names(lang_name) <- NULL # Strip name
+  lang_name
+}
+
+#' Function for Text Analysis Analytics View
+#' @export
+exp_textanal <- function(df, text,
+                         remove_punct = TRUE, remove_numbers = TRUE,
+                         stopwords_lang = NULL, stopwords = c(),
+                         hiragana_word_length_to_remove = 2,
+                         compound_tokens = NULL,
+                         cooccurrence_context = "window", # "document" or "window"
+                         cooccurrence_window = 1, # 5 is the quanteda's default, but narrowing it for speed of default run. 
+                         max_nrow = 50000,
+                         ...){
+
+  # Always put document_id to know what document the tokens are from
+  text_col <- tidyselect::vars_pull(names(df), !! rlang::enquo(text))
+  doc_id <- avoid_conflict(colnames(df), "document_id")
+  each_func <- function(df) {
+    # sample the data for performance if data size is too large.
+    sampled_nrow <- NULL
+    if (!is.null(max_nrow) && nrow(df) > max_nrow) {
+      # Record that sampling happened.
+      sampled_nrow <- max_nrow
+      df <- df %>% sample_rows(max_nrow)
+    }
+
+    # <Tokenizing code with quoanteda's default tokenizer.> TODO: Remove it when there is no need to keep it as a reference.
+    #
+    # This is SE version of dplyr::mutate(df, doc_id = row_number())
+    # df <- dplyr::mutate_(df, .dots=setNames(list(~row_number()),doc_id))
+    #
+    # textData <- df %>% dplyr::select(!!rlang::sym(text_col)) %>% dplyr::rename("text" = !!rlang::sym(text_col))
+    # # Create a corpus from the text column then tokenize.
+    # tokens <- quanteda::corpus(textData) %>%
+    #   quanteda::tokens(what = token, remove_punct = remove_punct, remove_numbers = remove_numbers,
+    #                    remove_symbols = remove_symbols, remove_twitter = remove_twitter,
+    #                    remove_hyphens = remove_hyphens, remove_separators = remove_separators,
+    #                    remove_url = remove_url) %>%
+    #   quanteda::tokens_wordstem()
+
+    tokenized <- tokenizers::tokenize_words(df[[text_col]], lowercase = TRUE, stopwords = NULL, strip_punct = remove_punct, strip_numeric = remove_numbers, simplify = FALSE)
+    names(tokenized) <- paste0("text", 1:length(tokenized)) # Add unique names so the list so that it can be passed to quanteda::tokens().
```

Comment:
Could you clean the "so the list so that" part? Do you mean "to the list so that"?

Reply:
Updated the comment.